### ISSUE TITLE
Two transaction commits are called during CreateItemFromDescription

### ISSUE THREAD

**Issue by kovax** [cristal-ise/cristal-ise#349]
The main reason for thsi issues, that  `CreateItemFromDescription` predefined step updates two Items. First of all it is called in a context of a Factory Item, therfore it stores the Event and Outcome of the activity. On the other end it creates a new Item by calling the `ItemImplemenetation.initialise()` method. This method is actually an implementation of a CORBA method, which means it will have its own transaction management, i.e. uses it own locker for transaction key, and call ClusterStorage.commit(). This is a problem because if the transcation is aborted during the ItemImplemenetation after the initialise() call, the data of the new Item will not be reverted.

This means If there was an error during CreateItemFromDescription, and it was after the call to ItemImplemenetation.initialise(), there will be an empty record about a new Item (UUID) in the cristal-ise database tables. This record was not added to the Lookup tables and reporting db, therefore it will not appear on any screen and will not prevent any future operations.

**Comment by kovax** [comment]
TransactionManager has a major limitation, it can only handle one Item within a single transaction. The solution is to create a new Transaction manager which delegates all calls and tranaction management to the underlying ClusterStorages. 

- [ ] Convert TranasactionManager to an interface
- [ ] Rename original TranasactionManager to SimpleTransactionManager
- [ ] Implement DelegatingTransactionManager

**Comment by kovax** [comment]
Current status. Runing test `RoutingScriptExecSpecs.'LoopSplit using RoutingScript and ActivityDataHelper'()` throws this exception:

```
org.cristalise.kernel.common.PersistencyException: IDL:org.cristalise.kernel/common/PersistencyException:1.0  Access denied: '/entity/29e43411-ede4-43ee-83b5-0a3ba6d85787' has been locked for writing by java.lang.Object@7292470c
	at org.cristalise.kernel.persistency.SimpleTransactionManager.getLockingTransaction(SimpleTransactionManager.java:241)
	at org.cristalise.kernel.persistency.SimpleTransactionManager.put(SimpleTransactionManager.java:197)
	at org.cristalise.kernel.lifecycle.instance.Activity.request(Activity.java:267)
	at org.cristalise.kernel.lifecycle.instance.Activity.request(Activity.java:195)
	at org.cristalise.kernel.lifecycle.instance.Workflow.requestAction(Workflow.java:135)
	at org.cristalise.dsl.test.builders.WorkflowTestBuilder.requestAction(WorkflowTestBuilder.groovy:242)
	at org.cristalise.dsl.test.builders.WorkflowTestBuilder.requestAction(WorkflowTestBuilder.groovy:214)
	at org.cristalise.kernel.test.lifecycle.instance.RoutingScriptExecSpecs.LoopSplit using RoutingScript and ActivityDataHelper(RoutingScriptExecSpecs.groovy:127)
```

**Comment by kovax** [comment]
TransactionManager was removed in 5.3.0 release to fix issue #426 and #427. This changes the scope of this bug entirely, which means that the branch feature/Issue349_TwoCommitsCreateItemFromDesc is deleted so all commits before this comment is irrelevant.
